controllers/machine_controller.go

```diff
@@ -183,9 +185,15 @@ func (r *MachineReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 		}
 	} else {
 		klog.Infof("Deleting node %q for machine %q", m.Status.NodeRef.Name, m.Name)
-		if err := r.deleteNode(ctx, cluster, m.Status.NodeRef.Name); err != nil && !apierrors.IsNotFound(err) {
-			klog.Errorf("Error deleting node %q for machine %q: %v", m.Status.NodeRef.Name, m.Name, err)
-			return ctrl.Result{}, err
+		waitErr := wait.PollImmediate(2*time.Second, 10*time.Second, func() (bool, error) {
+			if err = r.deleteNode(ctx, cluster, m.Status.NodeRef.Name); err != nil && !apierrors.IsNotFound(err) {
+				return false, nil
```

Comment:
Maybe we should record the error as `deleteNodeErr` and report that back to the user in the event instead of `waitErr` (I fear `waitErr` will be "timed out waiting for condition")?

Reply:
yes, `waitErr` will be "timed out waiting for condition"
do you mean doing this `deleteNodeErr = r.deleteNode` ? and keeping waitErr as it is?